podaac/subsetter/subset.py

```diff
@@ -819,8 +826,12 @@ def in_shape(lon, lat):
         point_in_shapefile = shapefile_df.contains(point)
         return point_in_shapefile.array[0]
 
+    dask = "forbidden"
+    if chunks:
+        dask = "allowed"
+
     in_shape_vec = np.vectorize(in_shape)
-    boolean_mask = xr.apply_ufunc(in_shape_vec, dataset[lon_var_name], dataset[lat_var_name])
+    boolean_mask = xr.apply_ufunc(in_shape_vec, dataset[lon_var_name], dataset[lat_var_name], dask=dask)
```

Comment:
Also curious about this line -- if there are no chunks (or auto) won't the data be numpy arrays (vs dask arrays) and dask won't be used anyways? I'm curious about why this is needed

Reply:
> Also curious about this line -- if there are no chunks (or auto) won't the data be numpy arrays (vs dask arrays) and dask won't be used anyways? I'm curious about why this is needed

When we use the chunk function a dask array is returned https://xarray.pydata.org/en/stable/generated/xarray.DataArray.chunk.html so if chunk is anything but {} we need to pass in dask=allowed in xr.apply_ufunc